raise-lock-pr/action.yaml

```diff
@@ -4,13 +4,20 @@ inputs:
   token:
     description: 'Github auth token with permissions to raise a PR'
     required: true
+  lock-settings:
```

Comment:
This is basically a hook to allow any cleanup that needs to be done before raising the PR. My specific use case came up after exhausting all other avenues I could think of.

After having some fun debugging deep into the gradle internals, I believe that there's a number of issues with how gradle handles the special `settings-gradle.lockfile`:
- there's a bug where several different configurations are persisted to the same file at the same time, causing it to be overwritten by the last entry. This [bug](https://github.com/gradle/gradle/issues/21101) refers to the same issue which seems to be confirmed but deprioritized. In our case, we're seeing the same symptom but with composite projects. The lock of the nested project is being written into the parent (for some gradle internal reasons I won't bore anyone with). This would be worth filing with gradle as a separate bug but given their responsiveness on the last one not sure we can count on that.
- Gradle hardcodes enabling locking in the settings file for the version catalog artifact. I don't see any way of overriding this. https://github.com/gradle/gradle/blob/b165da7de15e70afb6cac564bf4aadf16aa157b3/platforms/software/dependency-management/src/main/java/org/gradle/api/internal/catalog/DefaultVersionCatalogBuilder.java#L274
- Along similar lines, other tricks like moving the lock/adjusting the name don't appear to be available in settings. Adjusting the name of the catalog alters the configuration name such that it's different than the child's, but doesn't help because the whole file is overwritten due to the first point.

Altogether, that means the file will always be written when we `--write-locks` (as of gradle 8.4) and it'll always be wrong. The best work around I can come up with isn't great: given this workflow is the main mechanism of updating those locks, I'm going to have it by default remove the settings lock before raising the PR, and change the default behavior of the settings plugin to require a version for the catalog so that the lock becomes meaningless anyway. The build works just fine without the file. If updating locks locally rather than using this action, users will have to delete the file themselves. I'm intentionally _not_ adding the file to `.gitignore` however, since I don't want users to generate a bad lock file, not see it, and then be confused why their following build fails (since it won't match the expected state of the real catalogs dependencies).

Hope that all makes sense - dumb solution to a complicated problem!